admin/init.sls

```diff
@@ -27,9 +27,36 @@ UTC:
     - mode: 644
     - source: salt://{{ tpldir }}/files/hosts
 
-{% for ssh_user in admin.ssh_users %}
-sshkey-{{ ssh_user }}:
-  ssh_auth.present:
+# FIXME we should also explicitly AllowUsers usera userb userc
+/etc/sshd_config:
+  file.managed:
     - user: root
-    - source: salt://{{ tpldir }}/files/ssh/{{ ssh_user }}.pub
+    {% if grains['os'] == 'MacOS' %}
+    - group: wheel
+    {% elif grains['os'] == 'Ubuntu' %}
+    - group: root
+    {% endif %}
+    - mode: 644
+    - source: salt://{{ tpldir }}/files/sshd_config
+
+
+{% for ssh_user in admin.ssh_users %}
+{{ ssh_user }}:
+    user.present:
+        - empty_password: True
+        - optional_groups:
+            - wheel
+    ssh_auth.present:
+        - user: {{ ssh_user }}
+        - source: salt://{{ tpldir }}/files/ssh/{{ ssh_user }}.pub
+
 {% endfor %}
+
+# FIXME This is just as bad as all sharing root login.
+/etc/sudoers:
+    file.append:
```

Comment:
I've personally never been a fan of `file.append`, I prefer to use `file.managed` and control the whole contents instead of hoping that Salt is always smart enough to do the right thing with `file.append`.